src/scene/board-scene-controller.ts

```diff
@@ -105,13 +122,42 @@ export default class BoardSceneController extends SyncedSceneController {
         this.scene.updatePlayerInfo(uuid, player);
     }
 
+    public onPropertyBought(uuid: string, fields: number[], price: number) {
+        if (!this.aznopoly.isHost) return;
+
+        this.syncProxy.removeMoney(uuid, price);
+        this.syncProxy.addTiles(uuid, fields);
+    }
+
+    public onPayedRent(uuid: string, owner: string, rent: number) {
+        if (!this.aznopoly.isHost) return;
+
+        this.syncProxy.removeMoney(uuid, rent);
+        this.syncProxy.addMoney(owner, (rent * 0.5));
+    }
+
     public buyProperty(uuid: string) {
+        if (!this.aznopoly.isHost) return;
```

Comment:
how do you know this comes from the actual player that is eligible to buy to property?